doc/fundamentals/spouts.md

```diff
@@ -0,0 +1,291 @@
+# Spouts
+
+## Introduction
+
+Spouts are a way to get streaming data from any source into Pachyderm.
+To create a spout, you need three things
+
+1. A source of streaming data, such as Kafka, nifi, rabbitMQ, etc.
+2. A containerized server.
+3. A spout pipeline specification file that uses the container.
+
+The containerized server will do four things:
+
+- connect to your source of streaming data
+- read the data
+- package it into files 
+  in a `tar` stream
+- write that `tar` stream
+  to the named pipe `/pfs/out`
+
+
+In this document, 
+we'll take you through writing the server 
+with an example using Go code
```

Comment:
I know this is a bit late, but in the future, I really we should have a general policy of doing our examples (and especially docs) in Python. It's the lingua franca of the data world, it's explicitly designed to read as pseudocode, so anyone who knows any programming language should be able to understand it without any additional context, and everyone on Unix can run it without installing any additional infra

Reply:
Oh, this is my fault -- I wrote the kafka example used here in Go just since I'm familiar with Go and not Python.

Reply:
My plan is to write a use case example in Python and then make that the basis of a document, but the time crunch necessitated this approach.  Agree with the Python approach in the future.